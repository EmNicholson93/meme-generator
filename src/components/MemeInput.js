import React from 'react';
import PropTypes from 'prop-types';


function MemeInput({ head, imageSrc, foot, handleChange, handleSubmit }) {
  return (
    <>
      <form>
        <input name="head" value={head} placeholder="What should this meme read at the top?" onChange={(event) => handleChange(event)}></input>
        <input name="imageSrc" value={imageSrc} placeholder="What image would you like to use?" onChange={(event) => handleChange(event)}></input>
        <input name="foot" value={foot} placeholder="What should this meme read at the top?" onChange={(event) => handleChange(event)}></input>
      </form>
    </>
  );
}

MemeInput.propTypes = {
  head: PropTypes.string.isRequired,
  imageSrc: PropTypes.string.isRequired,
  foot: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
};

export default MemeInput;
