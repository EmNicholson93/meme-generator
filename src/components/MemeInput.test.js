import React from 'react';
import { shallow } from 'enzyme';
import MemeInput from './MemeInput';

describe('MemeInput component', () => {
  it('renders MemeInput', () => {
    const handleChange = jest.fn();
    const wrapper = shallow(<MemeInput head="hello" imgSrc="yolo" foot="feet" handleChange={handleChange} />);
    expect(wrapper).toMatchSnapshot();
  });
});
