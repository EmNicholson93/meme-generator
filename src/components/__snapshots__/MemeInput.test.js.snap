// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MemeInput component renders MemeInput 1`] = `
<Fragment>
  <form>
    <input
      name="head"
      onChange={[Function]}
      placeholder="What should this meme read at the top?"
      value="hello"
    />
    <input
      name="imageSrc"
      onChange={[Function]}
      placeholder="What image would you like to use?"
    />
    <input
      name="foot"
      onChange={[Function]}
      placeholder="What should this meme read at the top?"
      value="feet"
    />
  </form>
</Fragment>
`;
